temporalcli/client.go

```diff
@@ -12,47 +10,167 @@ import (
 
 	"go.temporal.io/api/common/v1"
 	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/contrib/envconfig"
 	"go.temporal.io/sdk/converter"
 	"go.temporal.io/sdk/log"
-
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
 )
 
 func (c *ClientOptions) dialClient(cctx *CommandContext) (client.Client, error) {
-	clientOptions := client.Options{
-		HostPort:  c.Address,
-		Namespace: c.Namespace,
-		Logger:    log.NewStructuredLogger(cctx.Logger),
-		Identity:  clientIdentity(),
-		// We do not put codec on data converter here, it is applied via
-		// interceptor. Same for failure conversion.
-		// XXX: If this is altered to be more dynamic, have to also update
-		// everywhere DataConverterWithRawValue is used.
-		DataConverter: DataConverterWithRawValue,
-	}
-
-	// API key
-	if c.ApiKey != "" {
-		clientOptions.Credentials = client.NewAPIKeyStaticCredentials(c.ApiKey)
+	if cctx.RootCommand == nil {
+		return nil, fmt.Errorf("root command unexpectedly missing when dialing client")
 	}
 
-	// Headers
+	// Load a client config profile
+	var clientProfile envconfig.ClientConfigProfile
+	if !cctx.RootCommand.DisableConfigFile || !cctx.RootCommand.DisableConfigEnv {
+		var err error
+		clientProfile, err = envconfig.LoadClientConfigProfile(envconfig.LoadClientConfigProfileOptions{
+			ConfigFilePath:    cctx.RootCommand.ConfigFile,
+			ConfigFileProfile: cctx.RootCommand.Profile,
+			DisableFile:       cctx.RootCommand.DisableConfigFile,
+			DisableEnv:        cctx.RootCommand.DisableConfigEnv,
+			EnvLookup:         cctx.Options.EnvLookup,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed loading client config: %w", err)
+		}
+	}
+
+	// To support legacy TLS environment variables, if they are present, we will
+	// have them force-override anything loaded from existing file or env
+	if !cctx.RootCommand.DisableConfigEnv {
+		oldEnvTLSCert, _ := cctx.Options.EnvLookup.LookupEnv("TEMPORAL_TLS_CERT")
```

Comment:
Mentioned in the other PR, but this sure seems like a good reason to export the constants from the SDK